components/sup/src/manager/service/mod.rs

```diff
@@ -732,11 +706,6 @@ impl Service {
         Ok(())
     }
 
-    fn run_hook(&mut self, hook: HookType) -> Result<()> {
-        self.hooks.compile(&self.service_group, &self.config);
-        self.hooks.try_run(hook, &self.service_group)
-    }
-
     /// Update our own service rumor with a new configuration from the packages exported
```

Comment:
If there are config files that depend on those values a reconfigure would be triggered on the next `tick`. That gives a 1 second upper bound to how stale the hooks can be.